src/notebook-jobs-panel.tsx

```diff
@@ -77,6 +77,7 @@ export class NotebookJobsPanel extends VDomRenderer<JobsModel> {
         <TranslatorContext.Provider value={this._translator}>
           {this.model.jobsView === 'CreateJob' && (
             <CreateJob
+              key={this.model.createJobModel.key}
```

Comment:
So, what does the special `key` prop do? Well, it actually triggers a full re-mount rather than a re-render, meaning all effect hooks are run. This is important because we are only fetching and setting the environment list on initial render, in an effect hook with no dependencies. Notice that `key` is only changed when `emptyCreateJobModel()` is called and used in the create job commands in `index.tsx`. This means that even if `<CreateJob />` is already being rendered by the notebook jobs panel, we will re-mount it to run all the effect hooks.

I understand the issue isn't immediately obvious. Compare and contrast the before and after.

Before:

https://user-images.githubusercontent.com/44106031/196565291-3dcdb5a4-cc3b-420e-ba41-4830b8f3f8d4.mov

After:

https://user-images.githubusercontent.com/44106031/196565455-a9614347-bf56-4b0c-9180-7489be907733.mov